### DISCUSSION TITLE
Is there a reason why the extension is not available for Safari?

### DISCUSSION THREAD

**Discussion by locksoft** [uBlockOrigin/uBlock-issues#2842] [Q&A] [answered]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

Sorry, I can't find a way to contact directly developers and I have not a reddit account. I used uBlock Origin on Chrome and it's honestly the only blocker that really works. I'm desperately looking for something similar for Safari with no luck.
Is there a reason why there's not a plugin for this browser?
Is it planned?
Thanks.

### A specific URL where the issue occurs.

www.apple.ocm

### Steps to Reproduce

I can't simply find the extension

### Expected behavior

Well, work like on Chrome

### Actual behavior

No extension available

### uBO version

no one

### Browser name and version

Safari 17.0

### Operating System and version

MacOS Sonoma

**Comment by ghost** [comment, marked as answer]
https://github.com/uBlockOrigin/uBOL-home/issues/52

**Reply by locksoft** [reply]
Thank you.

Too bad, as all ads blockers I've tried so far on Safari don't seem to have any positive impact on pages nags results.

**Reply by locksoft** [reply]
This is just an example of AdGuard vs. uBlock Origin.

AdGuard:
![AdGuard](https://github.com/uBlockOrigin/uBlock-issues/assets/9337221/8e80a668-774d-4f6d-b0f9-3aef8e7bf88e)

uBlock Origin:
![uBlock Origin](https://github.com/uBlockOrigin/uBlock-issues/assets/9337221/474ed6ba-a4e4-4544-afb1-72d9ec491b6a)

The difference is scary! :)

**Reply by ghost** [reply]
Good result, but I will not trust it anyway https://twitter.com/gorhill/status/1583581072197312512

**Reply by locksoft** [reply]
Yes, you're right. But more or less these two results reflect my experience. While uBlock blocks almost everything, AdGuard does nothing (I've just realized that, turning it on or off still gives the same result, so there may be another problem, maybe with Sonoma).

**Reply by ivansavra1** [reply]
@locksoft Hey, unfortunately there's still no uBlock Origin for Safari, but I wanted to recommend you Wipr, I've had a very good experience with it over the past years, it's pretty close to uBlock Origin, in fact, I generally forget I have it because I never see ads, and I rarely see cookie notices. On the test you shared, it scores 100 points, and a perfect score on another test I found:
<img width="1033" alt="Screenshot 2024-11-27 at 3 24 53 AM" src="https://github.com/user-attachments/assets/57bba6af-681a-4fd7-b6d9-7f83a429b618">
<img width="1006" alt="Screenshot 2024-11-27 at 3 25 06 AM" src="https://github.com/user-attachments/assets/8776cac9-0f14-4bed-a966-a9973d241825">

It's made by one developer, so there are no "Acceptable ads" like on Adblock Plus and other shady ones.

**Reply by freezer2022** [reply]
There is a beta of wBlock available since the month and a half:
- https://github.com/uBlockOrigin/uBOL-home/issues/52#issuecomment-2447368341
- https://www.reddit.com/r/macapps/comments/1g02q1p/calling_all_safari_users_lets_take_back_the_web/

**Reply by locksoft** [reply]
> @locksoft Hey, unfortunately there's still no uBlock Origin for Safari, but I wanted to recommend you Wipr, I've had a very good experience with it over the past years, it's pretty close to uBlock Origin, in fact, I generally forget I have it because I never see ads, and I rarely see cookie notices. On the test you shared, it scores 100 points, and a perfect score on another test I found:
> [...]
> It's made by one developer, so there are no "Acceptable ads" like on Adblock Plus and other shady ones.

Thank you Ivan. I see that there are two versions of Wipr in the Apple store: Wipr 1 and Wipr 2. Which one are you using? Do you know the differences?

@freezer2022 Thanks for the suggestion. I tested that extension but my adblock results didn't change at all. Still got 69 on the first test above and 7% on the second one by @ivansavra1.

**Reply by MasterKia** [reply]
@locksoft You should not solely trust that test.

**Reply by ivansavra1** [reply]
@locksoft Hey, so sorry for not answering before, I forgot. Good news is that, at the time, I was using Wipr 1, since then I decided to buy Wipr 2 mainly to support the sole developer, I think the price is reasonable, especially when other paid solutions are subscription-based (Like 1Blocker and AdGuard)

The main difference I noticed, is that YouTube ads are now blocked flawlessly, and Wipr 2 (Unlike the original) is an universal app, you buy it once and it covers iOS and macOS.

**Reply by ivansavra1** [reply]
> @locksoft You should not solely trust that test.

It's a good indicative, of course it's best to do your own testing to see, but synthetic tests/benchmarks are faster, and easier to do, since you get a "score" to estimate.
If you're aware of better adblocking testing tools, please share.

**Reply by MasterKia** [reply]
> It's a good indicative

No it's not. Even a PiHole with "proper" filters can get a full score.

And by the way that test site is now dead.

**Reply by ivansavra1** [reply]
> > It's a good indicative
> 
> No it's not. Even a PiHole with "proper" filters can get a full score.
> 
> And by the way that test site is now dead.

Technically speaking, maybe it's not foolproof, but in my own experience, these kind of tests were indicative (With the exception of Brave, which for some reason gets imperfect scores, with no noticeable difference in terms of ads)

In the context of this thread, I think it's more productive to make tests and take them into account (Knowing full well it's just an indication), instead of just criticizing their use without providing a comparable alternative. Always remember that not everyone has the time or knowledge to make in-depth comparisons to find the best adblocker, some of us want something with reproducible scores that you can just "Install, enable, and forget about it", and I think that's fine.

**Reply by Yuki2718** [reply]
The test was just totally broken `https://brave.com/blog/adblocker-testing-websites-harm-users/`. It's not at all indicative of anything  - even the linked article doesn't explain all.
